From 71c46f2c17503da291d5fbcf028246e43bb2570d Mon Sep 17 00:00:00 2001
From: Algernon <lixiaodong67@huawei.com>
Date: Tue, 20 Jun 2023 11:51:12 +0800
Subject: [PATCH 158/256] fix kabi broken due to change of struct sas_ha_struct

driver inclusion
category: bugfix
bugzilla: https://gitee.com/src-openeuler/kernel/issues/I8EKNE

----------------------------------------------------------------------------

Commit ("scsi: libsas: Add gfp_t flags parameter
to event notifications") changes some struct sas_ha_struct, so
we need to fix kabi broken problem.

Signed-off-by: YunYi Yang <yangyunyi2@huawei.com>
---
 include/scsi/libsas.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/scsi/libsas.h b/include/scsi/libsas.h
index 3978717229bf..7db79ca4037f 100644
--- a/include/scsi/libsas.h
+++ b/include/scsi/libsas.h
@@ -404,11 +404,15 @@ struct sas_ha_struct {
 				* their siblings when forming wide ports */
 
 	/* LLDD calls these to notify the class of an event. */
+#ifndef __GENKSYMS__
 	int (*notify_port_event)(struct asd_sas_phy *, enum port_event,
 				 gfp_t gfp_flags);
 	int (*notify_phy_event)(struct asd_sas_phy *, enum phy_event,
 				gfp_t gfp_flags);
-
+#else
+	int (*notify_port_event)(struct asd_sas_phy *, enum port_event);
+	int (*notify_phy_event)(struct asd_sas_phy *, enum phy_event);
+#endif
 	void *lldd_ha;		  /* not touched by sas class code */
 
 	struct list_head eh_done_q;  /* complete via scsi_eh_flush_done_q */
-- 
2.27.0

